DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UIDatePickerRecorder.swift

```diff
@@ -16,9 +16,9 @@ internal struct UIDatePickerRecorder: NodeRecorder {
 
     init(identifier: UUID) {
         self.identifier = identifier
-        self.wheelsStyleRecorder = WheelsStyleDatePickerRecorder(identifier: identifier)
-        self.compactStyleRecorder = CompactStyleDatePickerRecorder(identifier: identifier)
-        self.inlineStyleRecorder = InlineStyleDatePickerRecorder(identifier: identifier)
+        self.wheelsStyleRecorder = WheelsStyleDatePickerRecorder(identifier: UUID())
+        self.compactStyleRecorder = CompactStyleDatePickerRecorder(identifier: UUID())
+        self.inlineStyleRecorder = InlineStyleDatePickerRecorder(identifier: UUID())
```

Comment:
/suggestion We could make an init with default value instead:
```swift
init(identifier: UUID = UUID()) { }
```
---
It is still unclear when to create a recorder with new uuid and when to inject one. If I understand correctly, the fix in #1967 introduced another issue, the actual fix is in this PR in `UITabBarRecorder.swift`, right?

I guess it might be best to revert changes in #1967 to prevent any mis-usage in the future: expecting an `identifier` parameter could be error-prone, WDYT?

Reply:
The bottom line strategy is: recorders should have a stable and unique id. We should avoid creating temporary recorders inline inside of function.

Both changes are relevant. IMO it's better to keep explicit id (it makes you stop for a second).

I'll add a documentation within Recorder.